### PR TITLE
add Logger::get_effective_level().

### PR DESCRIPTION
**Pull request by fujitatomoya**
@clalancette @iuhilnehc-ynos could you do review when you have time?

### CODE REVIEW

File: rclcpp/include/rclcpp/logger.hpp [hunk 1]
```diff
@@ -170,6 +170,15 @@ class Logger
   RCLCPP_PUBLIC
   void
   set_level(Level level);
+
+  /// Get effective level for current logger.
```
Comment (iuhilnehc-ynos) [inline rclcpp/include/rclcpp/logger.hpp:174]: Users might not know the whole meaning of effective level here, especially for a logger that has ancestors.
https://github.com/ros2/rcutils/blob/90461677df74ed2dc3c5a4802a156a16f9082820/include/rcutils/logging.h#L416-L421 ?

File: rclcpp/test/rclcpp/test_logger.cpp [hunk 2]
```diff
@@ -160,6 +160,39 @@ TEST(TestLogger, set_level) {
...
+  logger.set_level(rclcpp::Logger::Level::Warn);
+  EXPECT_EQ(rclcpp::Logger::Level::Warn, logger.get_effective_level());
+
+  // error
+  logger.set_level(rclcpp::Logger::Level::Error);
+  EXPECT_EQ(rclcpp::Logger::Level::Error, logger.get_effective_level());
+
+  // fatal
+  logger.set_level(rclcpp::Logger::Level::Fatal);
+  EXPECT_EQ(rclcpp::Logger::Level::Fatal, logger.get_effective_level());
```
Comment (iuhilnehc-ynos) [inline rclcpp/test/rclcpp/test_logger.cpp:193]: How about adding some more tests about `rclcpp::get_logger("test_logger.child")`?

**Review by iuhilnehc-ynos — COMMENTED** [review]
two minor suggestions

File: rclcpp/include/rclcpp/logger.hpp (hunk 1 above)
Comment (fujitatomoya) [reply rclcpp/include/rclcpp/logger.hpp:174]: that is true, should be added to doc section.

File: rclcpp/test/rclcpp/test_logger.cpp (hunk 2 above)
Comment (fujitatomoya) [reply rclcpp/test/rclcpp/test_logger.cpp:193]: yeah that makes sense.

Comment (fujitatomoya) [reply rclcpp/test/rclcpp/test_logger.cpp:203]: addressed by https://github.com/ros2/rclcpp/pull/2141/commits/b185c701093ffbe3d711c61fe07794e7875bc6f2

File: rclcpp/include/rclcpp/logger.hpp (hunk 1 above)
Comment (fujitatomoya) [reply rclcpp/include/rclcpp/logger.hpp:174]: addressed by https://github.com/ros2/rclcpp/pull/2141/commits/b185c701093ffbe3d711c61fe07794e7875bc6f2

**Review by iuhilnehc-ynos — APPROVED** [review]

**Comment by fujitatomoya** [comment]
@clalancette i will go ahead to merge this.

**Pull request merged**